sdk/storage/Azure.Storage.Files.DataLake/samples/Sample02_Auth.cs

```diff
@@ -0,0 +1,198 @@
+            //
+            string accountName = StorageAccountName;
+            string accountKey = StorageAccountKey;
+            Uri serviceUri = StorageAccountBlobUri;
+
+            // Make StorageSharedKeyCredential to pass to the serviceClient
+            StorageSharedKeyCredential sharedKeyCredential = new StorageSharedKeyCredential(accountName, accountKey);
+
+            // Get a reference to a service Client
+            DataLakeServiceClient service = new DataLakeServiceClient(serviceUri, sharedKeyCredential);
+
+            // Make a service request to verify we've succesfully authenticated
+            await service.CreateFileSystemAsync("sample-filesystem-sharedkey");
+            await service.DeleteFileSystemAsync("sample-filesystem-sharedkey");
+        }
+
+        /// <summary>
+        /// Use a shared access signature to acces a Storage Account.
+        ///
+        /// A shared access signature (SAS) is a URI that grants restricted
+        /// access rights to Azure Storage resources. You can provide a shared
+        /// access signature to clients who should not be trusted with your
+        /// storage account key but to whom you wish to delegate access to
+        /// certain storage account resources. By distributing a shared access
+        /// signature URI to these clients, you can grant them access to a
+        /// resource for a specified period of time, with a specified set of
+        /// permissions.
+        /// </summary>
+        [Test]
+        public async Task SharedAccessSignatureAuthAsync()
+        {
+            // Create a service level SAS that only allows reading from service
+            // level APIs
+            AccountSasBuilder sas = new AccountSasBuilder
+            {
+                Protocol = SasProtocol.None,
+                Services = AccountSasServices.Blobs,
+                ResourceTypes = AccountSasResourceTypes.All,
+                StartsOn = DateTimeOffset.UtcNow.AddHours(-1),
+                ExpiresOn = DateTimeOffset.UtcNow.AddHours(1),
+                IPRange = new SasIPRange(IPAddress.None, IPAddress.None)
+            };
+            // Allow read access
+            sas.SetPermissions(AccountSasPermissions.List);
+
+            // Create a SharedKeyCredential that we can use to sign the SAS token
+            StorageSharedKeyCredential credential = new StorageSharedKeyCredential(StorageAccountName, StorageAccountKey);
+
+            // Build a SAS URI
+            UriBuilder sasUri = new UriBuilder(StorageAccountBlobUri);
+            sasUri.Query = sas.ToSasQueryParameters(credential).ToString();
+
+            // Create a client that can authenticate with the SAS URI
+            DataLakeServiceClient service = new DataLakeServiceClient(sasUri.Uri);
+
+            // Make a service request to verify we've succesfully authenticated
+            await service.GetFileSystemsAsync().ToListAsync();
```

Comment:
Nit - where does `ToListAsync` live?  Let's remove if that's part of our test code.

Reply:
It lives in TestExtensions and TestExtensions is used throughout a lot of tests (blobs, batch, files), so I made a new issue to address removing ToListAsync (#8587)

Reply:
Just to be clear - I only mean removing it from Samples, not our tests.  We shouldn't use test utilities in our samples (when we can help it) because it makes it harder for customers to just copy/paste the sample code into their own apps.  I think it's fine to come back to it later though since you've got a bug tracking it.